nncf/common/quantization/quantizer_propagation/solver.py

```diff
@@ -0,0 +1,1211 @@
+                 scope_overrides: Dict = None,
+                 global_constraints: Dict[QuantizerGroup, QuantizationConstraints] = None,
+                 additional_unified_scale_op_scopes: List[List[str]] = None,
+                 run_consistency_checks: bool = False,
+                 quantize_outputs: bool = False):
+        self.default_global_qconfig_list = default_qconfig_list
+        self._hw_config = hw_config  # type: HWConfig
+        self._debug_interface = debug_interface
+        self._propagation_strategy = propagation_strategy if propagation_strategy \
+            else QuantizerPropagationSolver.DEFAULT_PROPAGATION_STRATEGY  # TODO: determine from config
+        self._operator_quantization_trait_map = self.get_operator_quantization_traits_map()
+        self._operator_allowed_qconfigs_map = self._get_operator_qconfigs_map()
+        self._quantize_outputs = quantize_outputs
+        if quantizable_module_nodes is not None:
+            self._weight_quantizable_node_names_vs_qconfigs = {
+                x.node.node_name: x.qconfig_list for x in quantizable_module_nodes
+            }  # type: Dict[NNCFNodeName, List[QuantizerConfig]]
+        else:
+            self._weight_quantizable_node_names_vs_qconfigs = {}  # type: Dict[NNCFNodeName, List[QuantizerConfig]]
+        if scope_overrides is None:
+            self._scope_overrides = {}
+        else:
+            self._scope_overrides = scope_overrides  # type: Dict
+        self._global_constraints = global_constraints  # type: Dict['QuantizerGroup', 'QuantizationConstraints']
+        self._run_consistency_checks = run_consistency_checks
+
+        self._unified_scales_operation_set = set()
+        if self._hw_config is not None:
+            self._unified_scales_operation_set = self._hw_config.get_operations_with_unified_scales()
+
+        self._additional_unified_scale_op_scopes = additional_unified_scale_op_scopes
+
+        # Will handle the "wildcard" quantization situation for the time being
+        if default_qconfig_list is not None:
+            for op_meta, qconf_list in self._operator_allowed_qconfigs_map.items():
+                trait = self._operator_quantization_trait_map[op_meta]
+                if trait == QuantizationTrait.INPUTS_QUANTIZABLE:
+                    if HWConfig.is_qconf_list_corresponding_to_unspecified_op(qconf_list):
+                        self._operator_allowed_qconfigs_map[op_meta] = default_qconfig_list
+        self._active_propagating_quantizers_queue = deque()
+        self._finished_propagating_quantizers = []  # type: List[PropagatingQuantizer]
+        self._ignored_scopes = ignored_scopes
+        self._target_scopes = target_scopes
+        self._quantizers_waiting_for_branch_merge = QuantizersWaitingForMergeManager()
+
+        self._potential_quantizers = {}
+        self._num_potential_quantized_activations = 0
+
+    def run_on_ip_graph(self, ip_graph: InsertionPointGraph) -> QuantizationProposal:
```

Comment:
Could you please provide a description for the `ip_grap` parameter? Also, please add the `:return:` section.

Reply:
Done